### PR TITLE
Adapt Crate AMI Documentation

### PR DESCRIPTION
**Pull request by mikethebeer**
- documented how to adapt crate configuration
- documented data path behavior

### CODE REVIEW

File: docs/best_practice/crate_ami.txt [hunk 1]
```diff
@@ -12,7 +12,11 @@ convention is as follows:
   ``Crate-AWS-AMI-<VERSION>-<REVISION>``
 
 The placeholder ``<VERSION>`` is the Crate version in format x.y.z and
-``<REVISION`` describes its build revision.
+``<REVISION>`` describes its build revision. Crate AWS AMI is based on Amazon
+Linux (`Amazon Linux AMI`_) and it's pre-Installed packages are:
```
Comment (chaudum) [inline docs/best_practice/crate_ami.txt]: please reference the AMI id of the amazon linux ami

Comment (chaudum) [inline docs/best_practice/crate_ami.txt]: ~~it's pre-Installed packages~~
and comes with Java 8 (<exact package name>) and Crate (<exact package name>) pre-installed.

File: docs/best_practice/crate_ami.txt [hunk 2]
```diff
@@ -42,20 +46,41 @@ Launch the AMI
...
 .. code-block:: bash
 
     $ aws ec2 run-instances --image-id <AMI-ID> --count <NR-OF-INSTANCES> --instance-type <INSTANCE-TYPE> --user-data <USER-DATA>  --key-name <KEY-NAME>
 
+.. _cloud_init:
+
+Cloud-Init
+----------
+
+Crate AWS AMI uses `Cloud-Init`_ to setup all necessary configurations on
```
Comment (chaudum) [inline docs/best_practice/crate_ami.txt]: the ami creates a basic crate configuration at first launch (using cloud-init). this includes:

File: docs/best_practice/crate_ami.txt [hunk 3]
```diff
@@ -64,7 +89,38 @@ launching the instance.
...
+by using the User-Data script (see :ref:`user_data`). The following example
+shows how to set the ``minimum_master_nodes`` and gateway configuration setting
+which are essentail for a multi node setup.
+(:ref:`multi_node_setup`).
+
+.. code-block:: bash
+
+    #!/bin/bash
+    echo "
+    discovery.zen.minimum_master_nodes: 2
```
Comment (chaudum) [inline docs/best_practice/crate_ami.txt]: please explain, and make clear that this is an example when starting 3 nodes

**Comment by chaudum** [comment]
ack, pls merge
@ChrisChinchilla do you wanna take a look at the text and correct if necessary.

**Comment by ChrisChinchilla** [comment]
Made some small changes @mikethebeer and @chaudum

**Pull request merged**